### PR TITLE
[ans] add expiry check when get target address and primary name

### PR DESCRIPTION
**Pull request by angieyth**
as title

### CODE REVIEW

**Comment by 0xaptosj** [comment]
need to revert `sh_scripts/move_tests.sh`

File: core_v2/sources/v2_domain_e2e_tests.move [hunk 1]
```diff
@@ -12,6 +12,7 @@ module aptos_names_v2::v2_domain_e2e_tests {
     use std::signer;
     use std::string;
     use std::vector;
+    use aptos_names_v2::v2_domains::get_name_record_props;
```
Comment (0xaptosj) [inline core_v2/sources/v2_domain_e2e_tests.move:15]: remove this to use `v2_domains::get_name_record_props` to keep it consistent

File: core_v2/sources/v2_domains.move [hunk 2]
```diff
@@ -1107,7 +1121,7 @@ module aptos_names_v2::v2_domains {
         }
     }
 
-    public fun get_name_record_props_for_name(
+    public fun get_name_record_props(
```
[the remark this answers is not in the document]
Comment (0xaptosj) [reply core_v2/sources/v2_domains.move:1124]: +11111111

File: core_v2/sources/v2_domain_e2e_tests.move [hunk 3]
```diff
@@ -312,6 +313,69 @@ module aptos_names_v2::v2_domain_e2e_tests {
...
 
+    #[test(
+        router_signer = @router_signer,
+        aptos_names_v2 = @aptos_names_v2,
+        user = @0x077,
+        aptos = @0x1,
+        rando = @0x266f,
+        foundation = @0xf01d
+    )]
+    fun test_get_name_record_props_for_domain(
```
[the remark this answers is not in the document]
Comment (angieyth) [reply core_v2/sources/v2_domain_e2e_tests.move:323]: I'll split those into target address and expiration date. then create tests for subdomain as well

[the remark this answers is not in the document]
Comment (angieyth) [reply core_v2/sources/v2_domain_e2e_tests.move:323]: in next pr :P

File: core_v2/sources/v2_domain_e2e_tests.move [hunk 4]
```diff
@@ -312,6 +312,68 @@ module aptos_names_v2::v2_domain_e2e_tests {
...
+        // Register the domain
+        v2_test_helper::register_name(router_signer, user, option::none(), v2_test_helper::domain_name(), v2_test_helper::one_year_secs(), v2_test_helper::fq_domain_name(), 1);
+        v2_test_helper::set_target_address(user, v2_test_helper::domain_name(),option::none(),  user_addr);
+        let (expiration_sec, target_address) = v2_domains::get_name_record_props(option::none(), v2_test_helper::domain_name());
+        assert!(expiration_sec == v2_test_helper::one_year_secs(), 1);
+        assert!(target_address == option::some(user_addr), 2);
+
+        timestamp::update_global_time_for_test_secs(expiration_sec + 5);
+        let (_, target_address) = v2_domains::get_name_record_props(option::none(), v2_test_helper::domain_name());
+        assert!(target_address == option::none(), 3);
```
Comment (0xaptosj) [inline core_v2/sources/v2_domain_e2e_tests.move:344]: also use option::is_none to keep consistent

File: core_v2/sources/v2_domains.move [hunk 5]
```diff
@@ -1155,6 +1153,39 @@ module aptos_names_v2::v2_domains {
...
+            return (record.domain_name, option::none())
+        };
+        // option::get_with_default()
+        // let subdomain = *option::borrow(&subdomain_ext);
+        let subdomain_name = (option::borrow(&record.subdomain_ext)).subdomain_name;
+
+        return (record.domain_name, option::some(subdomain_name))
+    }
+
+    fun is_primary_name(
```
Comment (0xaptosj) [inline core_v2/sources/v2_domains.move:1170]: should we check expiration in here?

Comment (0xaptosj) [reply core_v2/sources/v2_domains.move:1170]: synced, get primary name checks expiration, no need here

**Review by 0xaptosj — APPROVED** [review]

File: core_v2/sources/v2_domains.move [hunk 6]
```diff
@@ -460,13 +450,29 @@ module aptos_names_v2::v2_domains {
...
+        record: &NameRecord,
+    ): bool acquires CollectionCapability, ReverseRecord{
+        if (option::is_some(&record.target_address)) {
+            let account_addr = *option::borrow(&record.target_address);
+            let reverse_record = borrow_global<ReverseRecord>(account_addr);
+            let maybe_reverse_record_addr = reverse_record.token_addr;
+            if (option::is_some(&maybe_reverse_record_addr)) {
+                let reverse_record_addr = *option::borrow(&maybe_reverse_record_addr);
+
+                return (get_token_addr_inline(record.domain_name, option::none()) == reverse_record_addr)
```
[the remark this answers is not in the document]
Comment (angieyth) [reply core_v2/sources/v2_domains.move]: I realized this is not a bug. this is for domain renewal only.

**Pull request merged**